src/create_agent/check_agent.jl

```diff
@@ -0,0 +1,50 @@
+"""
+
+Function for checking the structure of the agent
+"""
+function check_agent(agent::Agent)
+
+    if length(agent.shared_parameters) > 0
+
+        ## Check for the same derived parameter in multiple shared parameters 
+        #Get out the derived parameters of all shared parameters 
+        derived_parameters = [
+            parameter for list_of_derived_parameters in [
+                agent.shared_parameters[i].derived_parameters for
+                i in keys(agent.shared_parameters)
```

Comment:
hader personligt i'er og sådan, så ville skrive parameter_key eller noget.
Kan godt være det er bedre at ha det i to linjer, det er som du vil. 
Det er nice dog